### PR TITLE
Firewall Settings: Add Protect Flows

### PR DESCRIPTION
**Pull request by nateweller**
The purpose of this PR is to introduce Jetpack Protect as the primary method of activating and managing the Jetpack Firewall.

## Proposed changes:

* Jetpack Settings: Display an "activate" prompt for users who have neither the WAF nor Jetpack Protect activated. This leads them through the My Jetpack install/interstitial flow and lands on the firewall settings screen.
* Jetpack Settings: Display a link to the Jetpack Protect firewall settings screen for users who have Jetpack Protect already installed.
* Jetpack Settings: For users who already have the firewall active, and do not have Protect installed, a prompt to activate Protect is still shown. However, the user can dismiss this suggestion and continue using the now "legacy" settings UI.
* My Jetpack: Add a new `Protect_Firewall` product class, which extends the main `Protect` product. This is used to provide feature-tailored onboarding flows in My Jetpack, i.e. redirecting to the feature's tab post activation.
* My Jetpack: Fix the redirect handling in the `activate` `onSettled` method. 

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

p2-peb6dq-20F

## Does this pull request change what data or activity we track or use?

Yes, minor – adds "protect" as a possible event feature value to the existing upgrade banner tracking.

## Testing instructions:

* Start with Jetpack, and no Protect.
* Visit `/wp-admin/admin.php?page=jetpack#/settings`
* Test the activation flow. Validate you are redirected to the firewall tab after activating.
* Go back to the settings page once Protect is active, validate the settings screen redirects to Protect.
* Uninstall Protect.
* Go back to the settings page, validate it prompts you to activate Protect, but allows you to access the legacy settings UI.

## Screenshots

When Protect is installed, redirect the user to their Firewall dashboard:

Screenshot 2024-01-24 at 4 27 51 PM

If the firewall is disabled, run the feature activation through the Protect install/upgrade flow:

Screenshot 2024-01-22 at 5 14 00 PM

For users who already have the firewall enabled, prompt them to activate Protect, but allow them to access the "legacy" settings if they choose.

Screenshot 2024-01-24 at 4 14 21 PM

Screenshot 2024-01-24 at 4 15 12 PM

### CODE REVIEW

**Comment by miguelxpn** [comment]
I love the branch name!

File: projects/plugins/jetpack/_inc/client/state/recommendations/reducer.js [hunk 1]
```diff
@@ -516,7 +516,11 @@ export const isFeatureActive = ( state, featureSlug ) => {
 		case 'woocommerce':
 			return !! isPluginActive( state, 'woocommerce/woocommerce.php' );
 		case 'protect':
-			return !! isPluginActive( state, 'jetpack-protect/jetpack-protect.php' );
+			return (
+				!! isPluginActive( state, 'jetpack-protect/jetpack-protect.php' ) ||
+				!! isPluginActive( state, 'protect/jetpack-protect.php' ) ||
+				!! isPluginActive( state, 'jetpack-protect-dev/jetpack-protect.php' )
+			);
```
Comment (nateweller) [inline projects/plugins/jetpack/_inc/client/state/recommendations/reducer.js:523]: Added alternate plugin names here so the feature check works in testing and development environments.
- In typical production sites, the plugin will be `jetpack-protect/jetpack-protect.php`.
- In the monorepo docker setup, the plugin will be `protect/jetpack-protect.php` due to the "protect" plugin directory symlink.
- Protect may be active via Jetpack Beta, which is `jetpack-protect-dev/jetpack-protect.php`

File: projects/packages/my-jetpack/_inc/components/product-interstitial/index.jsx [hunk 2]
```diff
@@ -160,8 +162,8 @@ export default function ProductInterstitial( {
 					{ tiers && tiers.length ? (
 						<ProductDetailTable
 							slug={ slug }
-							clickHandler={ clickHandler }
```
Comment (nateweller) [inline projects/packages/my-jetpack/_inc/components/product-interstitial/index.jsx]: (Delete unused prop)

**Review by miguelxpn — DISMISSED** [review]

**Comment by nateweller** [comment]
Updated with a new custom redirect approach after some changes were made in My Jetpack before I could get this merged. Could use another top-to-bottom review, apologies! Thank you!

File: projects/packages/my-jetpack/_inc/components/product-interstitial/index.jsx [hunk 3]
```diff
@@ -14,6 +15,8 @@ import useProduct from '../../data/products/use-product';
 import useAnalytics from '../../hooks/use-analytics';
 import { useGoBack } from '../../hooks/use-go-back';
 import useMyJetpackNavigate from '../../hooks/use-my-jetpack-navigate';
+import { useProduct } from '../../hooks/use-product';
```
Comment (dkmyta) [inline projects/packages/my-jetpack/_inc/components/product-interstitial/index.jsx]: Looks like the import doesn't exist and is defined previously, I believe we can remove this and just default to using the original `useProduct` import? This should resolve the build and ESLint errors in the PR checks.

**Review by dkmyta — CHANGES_REQUESTED** [review]
If the WAF isn’t yet an active module (on initial activation of Jetpack on a clean install for example), we allow users to use legacy settings, do we want this or to only allow users who have previously activated the WAF module to have access?
Screenshot 2024-03-06 at 10 42 43 AM

I was not redirected back to the settings as expected after following the activation flow as described in step 3 of the testing instructions. Instead after connecting I was redirected to `/wp-admin/admin.php?page=jetpack#/dashboard`. 

After returning to the settings page, and clicking `Activate` from the Firewall section, I was routed to `#/add-firewall` but it was just a blank page with the following console warnings/errors:
```
Warning: Failed prop type: The prop `isFetching` is marked as required in `ProductDetailTable`, but its value is `undefined

Uncaught TypeError: setProduct is not a function
```

As an extra note, I think we should also have someone more familiar with the My Jetpack package give this a quick look after we approve - I see a review from Agora was requested but I am not sure they are responsible for this anymore.

**Comment by nateweller** [comment]
Ready for another review! PR description updated. The notes:

* New approach to accomplish a custom redirect for firewall-centric Protect activation flows, adding a child class that extends the `Protect` product with a custom return value for `get_manage_url()`.
* Includes an adjustment to the connection/checkout/redirect handling in My Jetpack to fix the redirect URL.

**Comment by nateweller** [comment]
Another update, some feedback on extending product classes: p1717089889789209-slack-CDLH4C1UZ

May need a different, different approach. I wanted to define this on the server side to keep things clean, but if we can't do that with the existing patterns then I start to lean towards coding in some sort of exception or override on the client side, specific to the Protect interstitial. Perhaps a `?feature=firewall` query parameter or similar that overrides the redirect URL of the Protect product when present.

**Pull request closed without merging**